src/Service.Tests/Unittests/ConfigValidationUnitTests.cs

```diff
@@ -332,20 +332,26 @@ public void TestAddingRelationshipWithDisabledGraphQL()
                 configValidator.ValidateRelationshipsInConfig(runtimeConfig, _metadataProviderFactory.Object));
-            Assert.AreEqual($"entity: {sampleRelationship.TargetEntity} is disabled for GraphQL.", ex.Message);
+            Assert.AreEqual($"Entity: {sampleRelationship.TargetEntity} is disabled for GraphQL.", ex.Message);
             Assert.AreEqual(HttpStatusCode.ServiceUnavailable, ex.StatusCode);
         }
 
         /// <summary>
-        /// Test method to check that an exception is thrown when LinkingObject was provided
-        /// while either LinkingSourceField or SourceField is null, and either targetFields or LinkingTargetField is null.
-        /// And the relationship is not defined in the database.
-        /// Also verify that after adding foreignKeyPair in the Database, no exception is thrown.
+        /// Test method to check that an exception is thrown in a many-many relationship (LinkingObject was provided)
+        /// while linkingSourceFields and sourceFields are null, or targetFields and linkingTargetFields are null,
+        /// and also the relationship is not defined in the database through foreign keys on the missing side of
+        /// fields in the config for the many-many relationship. That means if source and linking source fields are
+        /// missing that the foreign key information does not exist in the database for source entity to linking object,
+        /// and if target and linking target fields are missing that the foreign key information does not exist in the
+        /// database for the target entity to linking object.
+        /// Further verify that after adding said foreignKeyPair in the Database, no exception is thrown. This is because
+        /// once we have that foreign key information we can complete that side of the many-many relationship
+        /// from that foreign key.
         /// </summary>
-        [DataRow(new string[] { "id" }, null, new string[] { "num" }, new string[] { "book_num" }, "SampleEntity1", DisplayName = "LinkingSourceField is null")]
-        [DataRow(null, new string[] { "token_id" }, new string[] { "num" }, new string[] { "book_num" }, "SampleEntity1", DisplayName = "SourceField is null")]
-        [DataRow(new string[] { "id" }, new string[] { "token_id" }, new string[] { "num" }, null, "SampleEntity2", DisplayName = "LinkingTargetField is null")]
-        [DataRow(new string[] { "id" }, new string[] { "token_id" }, null, new string[] { "book_num" }, "SampleEntity2", DisplayName = "TargetField is null")]
+        [DataRow(null, null, new string[] { "targetFields" }, new string[] { "linkingTargetFields" }, "SampleEntity1",
+            DisplayName = "sourceFields and LinkingSourceFields are null")]
+        [DataRow(new string[] { "sourceFields" }, new string[] { "linkingSourceFields" }, null, null, "SampleEntity2",
+            DisplayName = "targetFields and LinkingTargetFields are null")]
```

Comment:
```suggestion
        [DataRow(null, null, new string[] { "targetField" }, new string[] { "linkingTargetField" }, "SampleEntity1",
            DisplayName = "sourceFields and LinkingSourceFields are null")]
        [DataRow(new string[] { "sourceField" }, new string[] { "linkingSourceField" }, null, null, "SampleEntity2",
            DisplayName = "targetFields and LinkingTargetFields are null")]
```

super nit: because you're only defining one field per array.